x-pack/platform/plugins/shared/streams/server/lib/rules/esql/register.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AlertInstanceContext, RuleTypeState } from '@kbn/alerting-plugin/server';
+import { alertFieldMap } from '@kbn/alerts-as-data-utils';
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/server';
+import { ESQL_RULE_TYPE_ID } from '@kbn/rule-data-utils';
+import { PersistenceAlertType } from '@kbn/rule-registry-plugin/server';
+import {
+  STREAMS_FEATURE_ID,
+  STREAMS_RULE_REGISTRATION_CONTEXT,
+} from '../../../../common/constants';
+import { getRuleExecutor } from './executor';
+import { EsqlRuleParams, esqlRuleParams } from './types';
+
+export function esqlRuleType(): PersistenceAlertType<
+  EsqlRuleParams,
+  RuleTypeState,
+  AlertInstanceContext,
+  'default'
+> {
+  return {
+    id: ESQL_RULE_TYPE_ID,
+    name: 'ES|QL Rule',
+    validate: {
+      params: {
+        validate: (object: unknown) => {
+          return esqlRuleParams.parse(object);
+        },
+      },
+    },
+    schemas: {
+      params: { type: 'zod', schema: esqlRuleParams },
+    },
+    defaultActionGroupId: 'default',
+    actionGroups: [
+      {
+        id: 'default',
+        name: 'Default',
+      },
+    ],
+    category: DEFAULT_APP_CATEGORIES.observability.id,
+    producer: STREAMS_FEATURE_ID,
+    solution: 'observability',
```

Comment:
is there a way around this?

Reply:
that's coming from the alerting framework, not sure I can change that and what's the implication of it.

Reply:
@elastic/response-ops any advice here? we'd like to not scope this to a solution

Reply:
At the moment, it will be used in a feature where users will be able to add an alerts table to a dashboard. Users, before adding the table, will have to select a solution. We do not like that categorization, but we are forced to do that because of the differences between how o11y and security authorize alerts. Security is using Kibana and ES privileges, o11y only Kibana. Do you plan to show the alerts generated by the `stream` in the UI, or will they be hidden?

Reply:
Ok. We may have to filter out the alerts generated by the `stream` rule type in the alerts table in dashboards. cc @umbopepato

Reply:
@cnasikas maybe your previous suggestion of introducing some flag on the rule registration to mark the rule as hidden/backend only might be needed

Reply:
I believe the same. Let me talk about that with the team. I will come back to you asap.

Reply:
Are we looking to exclude the streams significant events from the alerts table temporarily while this is in tech-preview, or is this a permanent behavior?

Reply:
@kobelb  This is more a permanent behavior. Maybe it will change but for now we don't expect to show them in alerts table.